enhancements/windows-containers/health-management.md

```diff
@@ -0,0 +1,258 @@
+* WMCD cleans up all changes it has made to a VM when ran with the `deconfigure` command.
+
+The Windows Node creation workflow for the end user should not be changed.
+
+### Services ConfigMap
+
+WMCO will create and maintain an immutable ConfigMap with the naming scheme:
+`services-<MajorVersion>-<MinorVersion>-<PatchVersion>-<Commit>`. For example, if WMCO was built from commit a7b5
+of WMCO version 5.0.0, the created ConfigMap would be named `config-5-0-0-a7b5`.
+
+The purpose of this ConfigMap is to provide WMCD with the specifications required to configure a Node. The ConfigMap
+will be watched via the existing ConfigMap controller. If it is created with incorrect values, WMCO will delete it.
+If it is deleted, WMCO will re-create it. If it is not present when WMCO is kicking off the configuration of a VM,
+it will be created.
+
+The services ConfigMap contains two keys: `services`, and `files`. The values for both keys are JSON objects.
+The services key contains all data required to configure the required Windows services on any VM that is to be
+added to the cluster as a Node. The proposed schema is as follows:
+```json
+[
+  {
+    "name": "name of the Windows service",
+    "command": "the command that will be ran",
+    "nodeVariablesToSubstitute": {
+      "name": "string within command field that will be substituted",
+      "jsonPathNodeObject": "value that will be substituted, taken from the provided jsonPath of the Node object"
+    },
+    "powershellVariablesToSubstitute": {
+      "name": "string within command field that will be substituted",
+      "path": "value that will be substituted, taken from the result of the powershell script at this path"
+    }
+  }
+]
+```
+
+The files key contains the path and checksum of files copied to the VM by WMCO. This will be used to validate that
+essential files have not been tampered with.
+```json
+[
+  {
+    "path": "filepath",
+    "checksum": "checksum of file"
+  }
+]
+```
+
+### Node configuration: WMCO responsibilities
+
+1) For a given Machine or BYOH VM, WMCO copies the payload and WMCD binary to the VM.
+2) WMCO runs WMCD with the bootstrap command, this results in the node object being created.
+3) WMCO annotates the Node object with the `desiredVersion` annotation.
+4) The value of this is set to the current WMCO version.
+5) WMCO starts WMCD as a service.
+6) Reconciliation can successfully return at this point.
+
+### Node configuration: WMCD responsibilities
+
+For node configuration WMCD will have two separate responsibilities:
+
+#### Bootstrap command
+
+When run with the bootstrap command WMCD will do the steps required to ensure that Node is created
+for the VM. This functionality is currently implemented within WMCB with the `init-kubelet` command.
+
+#### Running as a Windows service
+
+WMCD will have an informer for both ConfigMap and Node events. When it is detected that either the desired version
```

Comment:
- How will the daemon know the desired version?
- Why does need to know about the ConfigMap changes given it is going to be immutable for a particular version?